src/paginationCalls/pageDescriptor.ts

```diff
@@ -122,93 +112,49 @@ export class PageDescriptor implements Descriptor {
   asyncIterate(
     apiCall: GaxCall,
     request: RequestType,
-    options: CallSettings
+    options?: CallSettings
   ): AsyncIterable<{} | undefined> {
-    const iterable = this.createIterator();
-    const funcPromise =
-      typeof apiCall === 'function' ? Promise.resolve(apiCall) : apiCall;
-    funcPromise
-      .then((func: GaxCall) => {
-        this.makeCall(request, func, options);
-      })
-      .catch(error => {
-        throw new Error(error);
-      });
+    options = Object.assign({}, options, {autoPaginate: false});
+    const iterable = this.createIterator(apiCall, request, options);
     return iterable;
```

Comment:
So here we don't have to wait for params to resolve, when `asyncIterate` is called, we have the `apiCall` and neede `request, options`, is that correct?

Reply:
Yes, because `apiCall` is already a function and everything is synchronous.